### PR TITLE
Add Improved and Greater Lastwall Soup Consumables and Effects

### PR DESCRIPTION
**Pull request by avagdumortesin**
Fixes https://github.com/foundryvtt/pf2e/issues/2551.

Since there are no items in the book for these versions of the soup, I just created Consumable Effects for them, and linked them to the Gravelands Herbalist Feat. If it's preferred that I create copies of the Lastwall soup item, please let me know, and I'll do so.

### CODE REVIEW

**Comment by SpartanCPA** [comment]
Please make an improved and a greater variant of the soup consumable

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                        "undead"
+                    ]
+                },
+                "selector": "stealth",
+                "type": "item",
+                "value": 3
+            }
+        ],
+        "source": {
+            "value": ""
```
Comment (SpartanCPA) [inline packs/data/consumable-effects.db/effect-lastwall-soup-improved.json:32]:
```suggestion
            "value": "Pathfinder Lost Omens: Knights of Lastwall"
```

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                "type": "item",
+                "value": 4
+            },
+            {
+                "key": "TempHP",
+                "value": 10
+            }
+        ],
+        "source": {
+            "value": ""
```
Comment (SpartanCPA) [inline packs/data/consumable-effects.db/effect-lastwall-soup-greater.json:36]:
```suggestion
            "value": "Pathfinder Lost Omens: Knights of Lastwall"
```

**Review by SpartanCPA — CHANGES_REQUESTED** [review]
Please add source to the original effect as well

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            "initiative": null,
+            "value": 0
+        },
+        "target": null,
+        "tokenIcon": {
+            "show": true
+        },
+        "traits": {
+            "custom": "",
+            "rarity": "common",
```
Comment (SpartanCPA) [inline packs/data/consumable-effects.db/effect-lastwall-soup-improved.json:44]:
```suggestion
            "rarity": "uncommon",
```

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+    "_id": "neZPoQF4hW3A31dd",
+    "data": {
+        "badge": null,
+        "description": {
+            "value": "<p>Granted by @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth checks against undead</li>\n<li>10 temporary hit points</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth DCs against undead</li>\n</ul>"
```
Comment (SpartanCPA) [inline packs/data/consumable-effects.db/effect-lastwall-soup-greater.json:6]:
```suggestion
            "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup (Greater)]{Lastwall Soup (Greater)}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth checks against undead</li>\n<li>10 temporary hit points</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth DCs against undead</li>\n</ul>"
```

File: packs/data/feats.db/gravelands-herbalist.json [hunk 5]
```diff
@@ -24,9 +24,6 @@
             "value": [
                 {
                     "value": "Herbalist Dedication"
-                },
-                {
-                    "value": "trained in Religion"
```
Comment (SpartanCPA) [inline packs/data/feats.db/gravelands-herbalist.json:29]: Why was this removed? Trained in Religion *is* a requirement

Comment (avagdumortesin) [reply packs/data/feats.db/gravelands-herbalist.json:29]: It does not show Trained in Religion as a requirement for Gravelands Herbalist on AON https://2e.aonprd.com/Feats.aspx?ID=3614, which is where I was looking. I'll add it back.

File: packs/data/equipment.db/lastwall-soup-greater.json [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        },
+        "preciousMaterial": {
+            "value": null
+        },
+        "preciousMaterialGrade": {
+            "value": null
+        },
+        "price": {
+            "value": {
+                "gp": 3500
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/equipment.db/lastwall-soup-greater.json:61]: I looked at other alchemical items with a similar item level and matched it. Crafting requires a price for the check, and nothing in the Herbalist Archetype changes that. I can remove it, if you want.

File: packs/data/feats.db/gravelands-herbalist.json (hunk 5 above)
Comment (avagdumortesin) [reply packs/data/feats.db/gravelands-herbalist.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json (hunk 4 above)
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 7]
```diff
@@ -0,0 +1,51 @@
+{
+    "_id": "5o33sch67Z8j5Vom",
+    "data": {
+        "badge": null,
+        "description": {
+            "value": "<p>Granted by @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth checks against undead</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth DCs against undead</li>\n</ul>"
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-improved.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            "value": 1
+        },
+        "rules": [
+            {
+                "key": "FlatModifier",
+                "label": "Lastwall Soup (vs. undead)",
+                "predicate": {
+                    "all": [
+                        "undead"
+                    ]
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            "value": 1
+        },
+        "rules": [
+            {
+                "key": "FlatModifier",
+                "label": "Lastwall Soup (vs. undead)",
+                "predicate": {
+                    "all": [
+                        "undead"
+                    ]
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-improved.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+            "value": "<p>Granted by @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth checks against undead</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth DCs against undead</li>\n</ul>"
+        },
+        "duration": {
+            "expiry": "turn-start",
+            "sustained": false,
+            "unit": "hours",
+            "value": 1
+        },
+        "level": {
+            "value": 1
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-improved.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json (hunk 3 above)
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-improved.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+            "initiative": null,
+            "value": 0
+        },
+        "target": null,
+        "tokenIcon": {
+            "show": true
+        },
+        "traits": {
+            "custom": "",
+            "rarity": "common",
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+            "value": "<p>Granted by @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth checks against undead</li>\n<li>10 temporary hit points</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth DCs against undead</li>\n</ul>"
+        },
+        "duration": {
+            "expiry": "turn-start",
+            "sustained": false,
+            "unit": "hours",
+            "value": 1
+        },
+        "level": {
+            "value": 1
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json]: Fixed in commit [e58fa08](https://github.com/foundryvtt/pf2e/pull/2596/commits/e58fa08ddcd4be0ff493652b4f8a9085a789fd65)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json (hunk 2 above)
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json]: Fixed in [6ea844f](https://github.com/foundryvtt/pf2e/pull/2596/commits/6ea844f1a79caa3ce2cafba0b28c385303c78d53)

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json (hunk 1 above)
Comment (avagdumortesin) [reply packs/data/consumable-effects.db/effect-lastwall-soup-improved.json]: [6ea844f](https://github.com/foundryvtt/pf2e/pull/2596/commits/6ea844f1a79caa3ce2cafba0b28c385303c78d53)

File: packs/data/equipment.db/lastwall-soup-greater.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/equipment.db/lastwall-soup-greater.json:61]: Updated the items with notes about how the pricing was determined in [3d1c540](https://github.com/foundryvtt/pf2e/pull/2596/commits/3d1c540c37564a064f8b3c254afb8c9bd75501e3)

File: packs/data/equipment.db/lastwall-soup-improved.json [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+        },
+        "preciousMaterial": {
+            "value": null
+        },
+        "preciousMaterialGrade": {
+            "value": null
+        },
+        "price": {
+            "value": {
+                "gp": 160
```
[the remark this answers is not in the document]
Comment (avagdumortesin) [reply packs/data/equipment.db/lastwall-soup-improved.json:61]: Updated the items with notes about how the pricing was determined in [3d1c540](https://github.com/foundryvtt/pf2e/pull/2596/commits/3d1c540c37564a064f8b3c254afb8c9bd75501e3)

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+            "unit": "hours",
+            "value": 1
+        },
+        "level": {
+            "value": 9
+        },
+        "rules": [
+            {
+                "key": "FlatModifier",
+                "label": "Lastwall Soup (vs. undead)",
```
Comment (TikaelSol) [inline packs/data/consumable-effects.db/effect-lastwall-soup-greater.json:20]:
```suggestion
```

File: packs/data/consumable-effects.db/effect-lastwall-soup-greater.json [hunk 15]
```diff
@@ -0,0 +1,56 @@
+{
+    "_id": "neZPoQF4hW3A31dd",
+    "data": {
+        "badge": null,
+        "description": {
+            "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup (Greater)]{Lastwall Soup (Greater)}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth checks against undead</li>\n<li>10 temporary hit points</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+4 item bonus to Stealth DCs against undead</li>\n</ul>"
```
[the remark this answers is not in the document]
Comment (TikaelSol) [reply packs/data/consumable-effects.db/effect-lastwall-soup-greater.json:6]: Adding in the `origin:trait:undead` under the `any` predicate makes it work for DCs just fine.

File: packs/data/consumable-effects.db/effect-lastwall-soup-improved.json [hunk 16]
```diff
@@ -0,0 +1,52 @@
+{
+    "_id": "5o33sch67Z8j5Vom",
+    "data": {
+        "badge": null,
+        "description": {
+            "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup (Improved)]{Lastwall Soup (Improved)}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth checks against undead</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+3 item bonus to Stealth DCs against undead</li>\n</ul>"
```
Comment (TikaelSol) [inline packs/data/consumable-effects.db/effect-lastwall-soup-improved.json:6]:
```suggestion
            "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup (Improved)]{Lastwall Soup (Improved)}</p>\n<p>The creature gains a +3 item bonus to Stealth checks and DCs against undead.</p>"
```

File: packs/data/consumable-effects.db/effect-lastwall-soup.json [hunk 17]
```diff
@@ -1,6 +1,7 @@
 {
     "_id": "6A8jsLR7upLGuRiv",
     "data": {
+        "badge": null,
         "description": {
             "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup]{Lastwall Soup}</p>\n<p>Implemented effects:</p>\n<ul>\n<li>+2 item bonus to Stealth checks against undead</li>\n</ul>\n<p>Unimplemented effects:</p>\n<ul>\n<li>+2 item bonus to Stealth DCs against undead</li>\n</ul>"
```
Comment (TikaelSol) [inline packs/data/consumable-effects.db/effect-lastwall-soup.json:6]:
```suggestion
            "value": "<p>Granted by @Compendium[pf2e.equipment-srd.Lastwall Soup]{Lastwall Soup}</p>\n<p>The creature gains a +2 item bonus to Stealth checks and DCs against undead.</p"
```

File: packs/data/equipment.db/lastwall-soup-improved.json [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+        "consumableType": {
+            "value": "elixir"
+        },
+        "consume": {
+            "_deprecated": true,
+            "value": ""
+        },
+        "containerId": null,
+        "description": {
+            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Improved)]{Effect: Lastwall Soup (Improved)}</p>\n<hr />\n<p><em>Note: This item can only be made by characters with the @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist} feat. Additionally, Paizo did not provide a price, so we set it as appropriate for a level 10 consumable.</em></p>"
```
Comment (TikaelSol) [inline packs/data/equipment.db/lastwall-soup-improved.json:32]:
```suggestion
            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Improved)]{Effect: Lastwall Soup (Improved)}</p>\n<hr />\n<p><em>Note:  Paizo did not provide a price for this item, so we set it as appropriate for a level 10 consumable.</em></p>"
```
We don't want to reiterate rule info in our editorial notes.

File: packs/data/equipment.db/lastwall-soup-greater.json [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+        "consumableType": {
+            "value": "elixir"
+        },
+        "consume": {
+            "_deprecated": true,
+            "value": ""
+        },
+        "containerId": null,
+        "description": {
+            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead. In addition, you gain 10 temporary hit points that last for 10 minutes.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Greater)]{Effect: Lastwall Soup (Greater)}</p>\n<hr />\n<p><em>Note: This item can only be made by characters with the @Compendium[pf2e.feats-srd.Gravelands Herbalist]{Gravelands Herbalist} feat. Additionally, Paizo did not provide a price, so we set it as appropriate for a level 18 consumable.</em></p>"
```
Comment (TikaelSol) [inline packs/data/equipment.db/lastwall-soup-greater.json:32]:
```suggestion
            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead. In addition, you gain 10 temporary hit points that last for 10 minutes.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Greater)]{Effect: Lastwall Soup (Greater)}</p>\n<hr />\n<p><em>Note: Paizo did not provide a price for this item, so we set it as appropriate for a level 18 consumable.</em></p>"
```

File: packs/data/equipment.db/lastwall-soup-improved.json [hunk 20]
```diff
@@ -29,7 +29,7 @@
...
+            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Improved)]{Effect: Lastwall Soup (Improved)}</p>\n<hr />\n<p><em>Note:  Paizo did not provide a price for this item, so we set it as appropriate for a level 10 consumable.</em></p>"
```
Comment (SpartanCPA) [inline packs/data/equipment.db/lastwall-soup-improved.json:32]:
```suggestion
            "value": "<p><strong>Activate</strong> 1 minute (Interact)</p>\n<hr />\n<p>Crimson Reclaimers make this rich, hearty soup using herbs that baffle the senses of the undead. A bowl of Lastwall soup is as nourishing as a full meal. Gravelands Herbalists have spent time with the Knights of Lastwall, and have learned how to improve upon Lastwall soup.</p>\n<p>For 1 hour after consuming a bowl of improved Lastwall soup, you gain a +3 item bonus to Stealth checks and DCs against undead.</p>\n<hr />\n<p>@Compendium[pf2e.consumable-effects.Effect: Lastwall Soup (Improved)]{Effect: Lastwall Soup (Improved)}</p>\n<hr />\n<p><em>Note: Paizo did not provide a price for this item, so we set it as appropriate for a level 10 consumable.</em></p>"
```

**Review by SpartanCPA — APPROVED** [review]
